pkg/cli/clients/clients.go

```diff
@@ -227,8 +227,17 @@ type ApplicationsManagementClient interface {
 	// GetResourceProviderSummary gets the resource provider summary with the specified name in the configured scope.
 	GetResourceProviderSummary(ctx context.Context, planeName string, providerNamespace string) (ucp_v20231001preview.ResourceProviderSummary, error)
 
+	// CreateOrUpdateResourceType creates or updates a resource type in the configured scope.
+	CreateOrUpdateResourceType(ctx context.Context, planeName string, providerNamespace string, resourceTypeName string, resource *ucp_v20231001preview.ResourceTypeResource) (ucp_v20231001preview.ResourceTypeResource, error)
+
 	// DeleteResourceType deletes a resource type in the configured scope.
 	DeleteResourceType(ctx context.Context, planeName string, providerNamespace string, resourceTypeName string) (bool, error)
+
+	// CreateOrUpdateAPIVersion creates or updates an API version in the configured scope.
+	CreateOrUpdateAPIVersion(ctx context.Context, planeName string, providerNamespace string, resourceTypeName string, apiVersionName string, resource *ucp_v20231001preview.APIVersionResource) (ucp_v20231001preview.APIVersionResource, error)
+
+	// CreateOrUpdateLocation creates or updates a resource provider location in the configured scope.
+	CreateOrUpdateLocation(ctx context.Context, planeName string, providerNamespace string, locationName string, resource *ucp_v20231001preview.LocationResource) (ucp_v20231001preview.LocationResource, error)
```

Comment:
I wonder if an example in comments would help, since I am confused about the interpretations. If its just me, then we dont need it.  planeName = /plane/radius/local, providrNamespace = similar to Applications.Core, locationName = global ?

Reply:
I'll add some examples to the interface. If you look, there's a lot of different functions defined on this type, and they are consistent with each other.

I think this is a challenge generally. If you need a refresh on the concepts and overall design of the Radius API surface, the best place for that is the docs, not the code. If we're adding examples, it has to scale.

Reply:
Also, if a resource provider is available in a location, would it be useful to add that as an arg in `func (amc *UCPApplicationsManagementClient) CreateOrUpdateResourceType(ctx context.Context, planeName string, resourceProviderName string, resourceTypeName string, resource *ucpv20231001.ResourceTypeResource) (ucpv20231001.ResourceTypeResource, error)`

Reply:
What do you mean? That it would update both at once? PTAL at how this is handled by `rad resource-provider create`. Right now these are separate API calls, and the design is pretty reasonable IMO.